esy-lib/Tarball.re

```diff
@@ -87,6 +110,7 @@ let unpackWithTar = (~stripComponents=?, ~dst, filename) => {
   | Some(stripComponents) =>
     Fs.withTempDir(out => {
       let* () = unpack(out);
+      let* () = fixFilePermissionsAfterUnTar(out);
```

Comment:
The permission should definitely only change if needed, this is the case because someone may use the permission as a way to encode data.

Reply:
yarn internally uses tar-fs for untarring, So tar-fs does a bitwise or between the current & expected permissions
https://github.com/mafintosh/tar-fs/blob/master/index.js#L107
I also did it in a similar way

Reply:
Is that something special to tar-fs or is it also done by normal tar?

Reply:
Actually the line for extracting is this, https://github.com/mafintosh/tar-fs/blob/7ec11cb27f93948193770f32b4d820e2e7195715/index.js#L229 and there is an important, & umask , which on my node will be 0o022, but I don't know exactly what it is, on node it is 

https://nodejs.org/api/process.html#process_process_umask

Reply:
But overall I think we can do it, if yarn is doing it, so feel free to ignore it.

Reply:
Yeah yarn is also using the umask thing, so we just need to ensure that, and we're good to go.

Reply:
@EduardoRFS @ManasJayanth I did the part of doing `permissions & ~umask` can you'll have a look again?